UPGRADE-2.0.md

```diff
@@ -1,5 +1,126 @@
+```
+
+* API firewalls have been renamed and user checkers have been configured on firewalls
+  with `security.user_checker.chain.<firewall>` service:
+
+```diff
+#config/packages/security.yaml
+
+security:
+    firewalls:
+        admin:
+            ...
++           user_checker: security.user_checker.chain.admin
+-       new_api_admin_user:
++       api_admin:
+            ...
++           user_checker: security.user_checker.chain.api_admin
+-       new_api_shop_user:
++       api_shop:
+            ...
++           user_checker: security.user_checker.chain.api_shop
+        shop:
+            ...
++           user_checker: security.user_checker.chain.shop
+```
+
+To reflect these changes, the route configuration must be updated accordingly:
+
+```diff
+# config/routes/sylius_api.yaml
+
+sylius_api:
+    resource: "@SyliusApiBundle/Resources/config/routing.yml"
+-   prefix: "%sylius.security.new_api_route%"
++   prefix: "%sylius.security.api_route%"
+
+```
+
+* Routing changes (note that these shop routes are not localized with the prefix: /{_locale} configuration entry):
+
+```md
+#config/routes/sylius_shop.yaml
+
+sylius_shop_payum:
+-   resource: "@SyliusShopBundle/Resources/config/routing/payum.yml"
++   resource: "@SyliusPayumBundle/Resources/config/routing/integrations/sylius_shop.yaml"
+
+sylius_payment_notify:
++   resource: "@SyliusPaymentBundle/Resources/config/routing/integrations/sylius.yaml"
+
+```
+
+* Bundle configuration changes:
+
+```md
+#config/bundles.php
+
+<?php
+
+return [
+-   Sylius\Calendar\SyliusCalendarBundle::class => ['all' => true],
+-   winzou\Bundle\StateMachineBundle\winzouStateMachineBundle::class => ['all' => true],
+-   Bazinga\Bundle\HateoasBundle\BazingaHateoasBundle::class => ['all' => true],
+-   JMS\SerializerBundle\JMSSerializerBundle::class => ['all' => true],
+-   FOS\RestBundle\FOSRestBundle::class => ['all' => true],
+-   ApiPlatform\Core\Bridge\Symfony\Bundle\ApiPlatformBundle::class => ['all' => true],
+-   SyliusLabs\Polyfill\Symfony\Security\Bundle\SyliusLabsPolyfillSymfonySecurityBundle::class => ['all' => true],
++   ApiPlatform\Symfony\Bundle\ApiPlatformBundle::class => ['all' => true],
++   Sylius\TwigHooks\SyliusTwigHooksBundle::class => ['all' => true],
++   Symfony\UX\TwigComponent\TwigComponentBundle::class => ['all' => true],
++   Symfony\UX\StimulusBundle\StimulusBundle::class => ['all' => true],
++   Symfony\UX\LiveComponent\LiveComponentBundle::class => ['all' => true],
++   Symfony\UX\Autocomplete\AutocompleteBundle::class => ['all' => true],
+];
+
+* New Symfony/Messenger transports for handling payment requests have been added. 
+Therefore, you need to add the following configuration to your .env file:
+```md
+###> symfony/messenger ###
+...
+SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_DSN=doctrine://default?queue_name=payment_request
+SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_FAILED_DSN=doctrine://default?queue_name=payment_request_failed
```

Comment:
```suggestion
SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_DSN=sync://
SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_FAILED_DSN=sync://
```
Async payment request can't be supported by Sylius right now, I made a PR (already merged) about this default value.